unittests/test_schedulers.py

```diff
@@ -123,6 +123,20 @@ def prepare_job(job, command='hostname',
         )
 
 
+def get_sleep_pid(job, attempts=3):
+    # Try reading the pid of spawned sleep, until a valid value is retrieved
+    for i in range(attempts):
```

Comment:
Is there a reason for not trying indefinitely? There is no way that you will not get it, right?

Reply:
I don't try indefinetely to be on the safe side so that it does not run forever